### PR TITLE
If applied, this commit will format Line items price in order api

### LINKED ISSUE #27911: REST-API Orders line_item price inconsistency when quantity > 1
Please provide us with the information requested in this bug report. Without these details, we won't be able to fully evaluate this issue. 
Bug reports lacking detail, or for any other reason than to report a bug, may be closed without action.

**Prerequisites (mark completed items with an [x]):**
- [x ] I have have carried out troubleshooting steps and I believe I have found a bug.
- [ x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
On line 172 in /woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php
`$data['price'] = $item->get_quantity() ? $item->get_total() / $item->get_quantity() : 0;`
the mathematical division in certain cases creates an inconsistent result. This code is inherited in V3. I am using V3

price in database = $14.31
qty ordered = 10
line_item subtotal (no tax) = 143.10

Rest Api response
'price':  '14.309999999999999',

**Expected behavior**
'price': '14.31'

**Actual behavior**
Rest Api response
'price':  '14.309999999999999',

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:

I am patching this code in production for now which fixes it temporarily. 
`$data['price'] = $item->get_quantity() ? wc_format_decimal( $item->get_total() / $item->get_quantity() ): 0;`
I am unsure if they WooCommerce team had a specific reason for allowing the precision of this float to go beyond two decimal places.

**Screenshots**
Screen Shot 2020-10-08 at 6 23 18 PM

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use the [WooCommerce System Status Report](https://docs.woocommerce.com/document/understanding-the-woocommerce-system-status-report/) to help us evaluate the issue. 
Without this report we won't be able to fully evaluate this issue.
<details>
```
The System Status Report is found in your WordPress admin under **WooCommerce > Status**. 
Please select “Get system report”, then “Copy for support”, and then paste it here.
```
</details>

### PR DESCRIPTION
**Pull request by linqiuyu**
When buying three products at price is 17.17. request /wp-json/wc/v3/orders api will get line_items prices is 17.169999999999998, this is because floating point numbers have limited precision.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

1. add a product price is 17.17
2. buy 3 this product
3. get request /wp-json/wc/v3/orders api

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Format price decimal places correctly in the order API.

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php [hunk 1]
```diff
@@ -169,7 +169,7 @@ protected function get_order_item_data( $item ) {
 		// Add SKU and PRICE to products.
 		if ( is_callable( array( $item, 'get_product' ) ) ) {
 			$data['sku']   = $item->get_product() ? $item->get_product()->get_sku() : null;
-			$data['price'] = $item->get_quantity() ? $item->get_total() / $item->get_quantity() : 0;
+			$data['price'] = $item->get_quantity() ? (float) number_format( $item->get_total() / $item->get_quantity(), wc_get_price_decimals(), '.', '' ) : 0;
```
Comment (jeffstieler) [inline plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]: Please use the `wc_format_decimal()` method here. See: https://github.com/woocommerce/woocommerce/pull/31593/files#diff-89305498bc7a59db846649f3b8e4e45708eba2903d252d5dbac5b2ef5c1dcfe3R165 for an example.

**Review by jeffstieler — CHANGES_REQUESTED** [review]
Hi @linqiuyu,

Nice catch - we just need to update the function used to format the decimal value.

Thank you for your contribution to WooCommerce!

File: plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php (hunk 1 above)
Comment (linqiuyu) [reply plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]: thank you, I modified my code.

**Review by jeffstieler — APPROVED** [review]
Looks good. Thank you!

_Just noting here that I ran the API tests locally to confirm_

**Pull request merged**